### PR TITLE
Add Introducing the Kubernetes Agent blog post

### PR DESCRIPTION
**Pull request by APErebus**
# About

This is a post about... the introduction of the new Kubernetes agent

# Release

This will be rolled out to cloud customers shortly, and will be in the next full release of self-hosted, I.e. 2024.2.

# Blog image idea

Something with octopus and kubernetes would be great. We do have a cool new Octopus Agent octo-person (:octo-agent:) that we might be able to use?

# Before you submit your post

Check everything off this list before you submit your post for review. 

- [x] I'm familiar with the [Octopus voice and style](https://www.octopus.design/932c0f1a9/p/13df3a-voice-and-style), the [guide for blog content basics](https://www.octopus.design/932c0f1a9/p/901d2a-blog-content), and [writing educational content](https://www.octopus.design/932c0f1a9/p/137b06-educational-content---how-we-write-about-what-we-do).
- [ ] I've checked the content using Grammarly (Octopus has a corporate license) and the [Hemingway app](https://hemingwayapp.com/) to remove typos, grammatical errors, and passive voice. 
- [x] The PR build passes validation, and if it doesn't, I've checked the [common validations errors](https://www.octopus.design/932c0f1a9/p/901d2a-blog-content/t/817249) and none of those apply.
- [x] The images I've included follow the [rules for working with images](https://www.octopus.design/932c0f1a9/p/5061d7-working-with-images).
- [x] I've added alt text for all the images I've included (alt text describes an image to people unable to see it, 125 characters max)
- [x] The draft is complete and this post is ready to be reviewed. (Please mark the post 'ready for review' at the bottom of the PR.)
- [x] I've dropped a note in Slack in #topic-content-marketing to let Tegan Ali know my post is ready for review. (No need to fill out a content brief.)
- [ ] I'll draft social media posts for Twitter and LinkedIn. I'll post them for review in #topic-social-media after the post has been edited or after the PR has been merged.

## How to review this PR

Is the 1st section too dense (the why) compared to the second section (the what)?

## Publication date

This is to be published to line up with the next newsletter

### CODE REVIEW

File: blog/2024-05/kubernetes-agent/index.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+tags: 
+  - Product
+  - Kubernetes
+  - AWS
+  - Azure
+  - Google
+---
+Kubernetes is rapidly becoming the dominant platform for hosting and running applications. At Octopus Deploy, we want to provide a best-in-class experience for deploying applications to Kubernetes.
+
+To make deployments to Kubernetes simpler, faster and safer, we are introducing a new deployment target the Kubernetes agent.
```
Comment (scme0) [inline blog/2024-05/kubernetes-agent/index.md:20]:
```suggestion
To make deployments to Kubernetes simpler, faster and safer, we are introducing a new deployment target: the Kubernetes agent.
```

I feel like it needs some kind of punctuation here. Not sure what though.
or maybe `a new deploment target called the Kubernetes agent.`?

Also did we agree on lower case a for agent?

File: blog/2024-05/kubernetes-agent/index.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+The Kubernetes agent is a small, lightweight application that you install into your Kubernetes cluster. Once installed, it registers itself with Octopus Server as a new deployment target, allowing you to deploy your applications and manifests into that cluster, without the need for workers, external credentials or custom tooling.
+
+The Kubernetes agent is available in Early Access preview in Octopus Cloud now.
+
+## Why did we build the agent?
+
+Octopus Deploy already has the ability to deploy to Kubernetes clusters via the Kubernetes API deployment target. This target connects to the target cluster, executing authenticated kubectl commands against the cluster.
+
+We have heard from customers that there is some limitations with the Kubernetes API target.
```
Comment (scme0) [inline blog/2024-05/kubernetes-agent/index.md:30]:
```suggestion
We have heard from customers that there are some limitations with the Kubernetes API target. 
```

File: blog/2024-05/kubernetes-agent/index.md [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+To execute a deployment to a Kubernetes API target, it requires an Octopus Worker with all the correct tooling. 
+In Octopus Cloud, we provide the Octopus Cloud Dynamic Workers, but when self-hosting Octopus Deploy, it requires either executing work on the Octopus Server machine or creating and managing an Octopus Worker.
+
+### Tooling
+
+As the Kubernetes API target uses kubectl to perform its deployments, it requires the correct tooling. This includes authentication plugins for cloud providers and other tools such as Helm. This becomes difficult to manage as versions change or when deploying to different clusters with different versions.
+
+### Authentication
+
+The Kubernetes API target requires authentication with the cluster to deploy. This authentication can be very complicated depending on the target cluster or hosting platform. These authentication credentials must be added to Octopus Deploy, making securing and automation difficult.
```
Comment (scme0) [inline blog/2024-05/kubernetes-agent/index.md:43]:
```suggestion
The Kubernetes API target requires authentication with the cluster to deploy. This authentication can be very complicated depending on the target cluster or hosting platform. These authentication credentials must be added to Octopus Deploy, making security and automation difficult.
```

File: blog/2024-05/kubernetes-agent/index.md [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+
+The Kubernetes agent requires a shared filesystem, so you have the option of either providing an existing storage class or using a default Network File System (NFS) storage pod. The NFS storage pod is an easy, low-configuration storage option that doesn’t need extra volumes or storage classes to be configured in the cluster first. To use NFS storage, you must install an extra helm chart.
+
+![Kubernetes agent wizard NFS driver](kubernetes-agent-wizard-nfs.png "width=500")*The NFS driver needs to be installed first*
+
+At the end of the wizard, Octopus Deploy will generate a Helm command that you copy and paste into a terminal connected to the target cluster. Once executed, Helm will install all the required resources and start the agent.
+
+![Kubernetes agent helm command](kubernetes-agent-wizard-helm-command.png "width=500")*Executing this helm command will install the agent into your cluster*
+
+Octopus Server will start waiting for the agent to establish a connection and perform a health check. Once successful, the Kubernetes agent target is ready for use!
```
Comment (scme0) [inline blog/2024-05/kubernetes-agent/index.md:83]: Maybe something like this?
```suggestion
If left open, the installation dialog will wait for the agent to establish a connection and perform a health check. Once successful, the Kubernetes agent target is ready for use!
```

**Review by scme0 — COMMENTED** [review]
Nice, looks good and reads well in my (non-copywriter) opinion! I've added a couple of comments which you can take or leave 👍

Do we need to add images in the top section for `metaImage`, `bannerImage` and `bannerImageAlt`?

I'll leave final approval up to Tegan

File: blog/2024-05/kubernetes-agent/index.md [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+## Why did we build the agent?
+
+Octopus Deploy already has the ability to deploy to Kubernetes clusters via the Kubernetes API deployment target. This target connects to the target cluster, executing authenticated kubectl commands against the cluster.
+
+We have heard from customers that there is some limitations with the Kubernetes API target. 
+
+### Workers
+
+To execute a deployment to a Kubernetes API target, it requires an Octopus Worker with all the correct tooling. 
+In Octopus Cloud, we provide the Octopus Cloud Dynamic Workers, but when self-hosting Octopus Deploy, it requires either executing work on the Octopus Server machine or creating and managing an Octopus Worker.
```
Comment (zentron) [inline blog/2024-05/kubernetes-agent/index.md:35]:
```suggestion
In Octopus Cloud, we provide the Octopus Cloud Dynamic Workers, but when self-hosting Octopus Deploy, it requires either executing work on the Octopus Server machine or creating and managing an [Octopus Tentacle](https://octopus.com/docs/infrastructure/deployment-targets/tentacle) installation running as an Octopus Worker.
```
I just added some clarity on what it means to run an Octopus Worker, requiring the running of a whole tentacle

File: blog/2024-05/kubernetes-agent/index.md [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+
+As the agent is already running inside the target cluster, it no longer needs authentication credentials to the cluster to perform deployments. It can leverage the in-cluster authentication support of Kubernetes to execute deployments using Kubernetes Service Accounts and Kubernetes RBAC local to the cluster.
+
+### Cluster-aware tooling
+
+As the agent is running in the cluster, it can retrieve the cluster's version and correctly use tooling that is specific to that version. The tooling required is also drastically reduced as there are no longer any requirements for custom authentication plugins.
+
+## How does the agent work?
+
+The Kubernetes agent builds on top of Octopus Tentacle, allowing for deployment scripts to be executed from Octopus Server. Where Octopus Tentacle executes scripts in local shells (via powershell or bash), the Kubernetes agent can take advantage of the Kubernetes cluster to execute deployment scripts inside of short-lived Kubernetes pods. This gives the agent the ability to scale its workloads independently of the main agent pod.
```
Comment (zentron) [inline blog/2024-05/kubernetes-agent/index.md:61]:
```suggestion
The Kubernetes agent builds on top of Octopus Tentacle, allowing for deployment scripts to be executed from Octopus Server. Whereas Octopus Tentacle executes scripts in local shells (via powershell or bash), the Kubernetes agent can take advantage of the Kubernetes cluster to execute deployment scripts inside of short-lived Kubernetes pods. This gives the agent the ability to scale its workloads independently of the main agent pod.
```

**Comment by zentron** [comment]
Looks good, I'd consider adding a few more links to docs when introducing some of the concepts but the general simple flow seems to work.

File: blog/2024-05/kubernetes-agent/index.md (hunk 1 above)
Comment (APErebus) [reply blog/2024-05/kubernetes-agent/index.md:20]: > Also did we agree on lower case a for agent?

Yep

> I feel like it needs some kind of punctuation here. Not sure what though.
> or maybe a new deployment target called the Kubernetes agent.?

Yeah, I had a similar thought

@TeganAli Any ideas?

Comment (TeganAli) [reply blog/2024-05/kubernetes-agent/index.md:20]:
```suggestion
To make your deployments to Kubernetes simpler, faster, and safer, we're introducing a new deployment target called the Kubernetes agent.
```

**Review by TeganAli — APPROVED** [review]

**Pull request merged**